graylog2-server/src/main/java/org/graylog2/dashboards/DashboardImpl.java

```diff
@@ -72,6 +79,49 @@ public void setDescription(String description) {
         this.fields.put(FIELD_DESCRIPTION, description);
     }
 
+    @Override
+    public List<WidgetPosition> getPositions() {
+        final BasicDBObject positions = (BasicDBObject) fields.get(DashboardImpl.EMBEDDED_POSITIONS);
+        if (positions == null) {
+            return Collections.emptyList();
+        }
+
+        final List<WidgetPosition> result = new ArrayList<>(positions.size());
+        for ( String positionId : positions.keySet() ) {
+            final BasicDBObject position = (BasicDBObject) positions.get(positionId);
+            final int width = parseInt(position.get("width").toString());
+            final int height = parseInt(position.get("height").toString());
+            final int col = parseInt(position.get("col").toString());
+            final int row = parseInt(position.get("row").toString());
+            final WidgetPosition widgetPosition = WidgetPosition.builder()
+                    .id(positionId)
+                    .width(width)
+                    .height(height)
+                    .col(col)
+                    .row(row)
+                    .build();
+            result.add(widgetPosition);
+        }
+        return result;
+    }
+
+    @Override
+    public void setPositions(List<WidgetPosition> widgetPositions) {
+        if (widgetPositions == null) {
+            return;
+        }
+        final Map<String, Map<String, Integer>> positions = new HashMap<>(widgetPositions.size());
+        for (WidgetPosition widgetPosition : widgetPositions) {
+            Map<String, Integer> position = new HashMap(4);
+            position.put("width", widgetPosition.width());
+            position.put("height", widgetPosition.height());
+            position.put("col", widgetPosition.col());
+            position.put("row", widgetPosition.row());
+            positions.put(widgetPosition.id(), position);
+        }
+        getFields().put(DashboardImpl.EMBEDDED_POSITIONS, positions);
```

Comment:
What would happen if `getFields()` returns `null`? (`fields` can be set directly using the constructor)